packages/cli/src/UserManagement/controllers/PasswordResetController.ts

```diff
@@ -0,0 +1,211 @@
+
+@RestController()
+export class PasswordResetController {
+	/**
+	 * Send a password reset email.
+	 *
+	 * Authless endpoint.
+	 */
+	@Post('/forgot-password')
+	async forgotPassword(req: PasswordResetRequest.Email) {
+		if (config.getEnv('userManagement.emails.mode') === '') {
+			Logger.debug('Request to send password reset email failed because emailing was not set up');
+			throw new ResponseHelper.ResponseError(
+				'Email sending must be set up in order to request a password reset email',
+				undefined,
+				500,
+			);
+		}
+
+		const { email } = req.body;
+
+		if (!email) {
+			Logger.debug(
+				'Request to send password reset email failed because of missing email in payload',
+				{ payload: req.body },
+			);
+			throw new ResponseHelper.ResponseError('Email is mandatory', undefined, 400);
+		}
+
+		if (!validator.isEmail(email)) {
+			Logger.debug(
+				'Request to send password reset email failed because of invalid email in payload',
+				{ invalidEmail: email },
+			);
+			throw new ResponseHelper.ResponseError('Invalid email address', undefined, 400);
+		}
+
+		// User should just be able to reset password if one is already present
+		const user = await Db.collections.User.findOne({ email, password: Not(IsNull()) });
+
+		if (!user || !user.password) {
+			Logger.debug(
+				'Request to send password reset email failed because no user was found for the provided email',
+				{ invalidEmail: email },
+			);
+			return;
+		}
+
+		user.resetPasswordToken = uuid();
+
+		const { id, firstName, lastName, resetPasswordToken } = user;
+
+		const resetPasswordTokenExpiration = Math.floor(Date.now() / 1000) + 7200;
+
+		await Db.collections.User.update(id, { resetPasswordToken, resetPasswordTokenExpiration });
+
+		const baseUrl = getInstanceBaseUrl();
+		const url = new URL(`${baseUrl}/change-password`);
+		url.searchParams.append('userId', id);
+		url.searchParams.append('token', resetPasswordToken);
+
+		try {
+			const mailer = await UserManagementMailer.getInstance();
+			await mailer.passwordReset({
+				email,
+				firstName,
+				lastName,
+				passwordResetUrl: url.toString(),
+				domain: baseUrl,
+			});
+		} catch (error) {
+			void InternalHooksManager.getInstance().onEmailFailed({
+				user_id: user.id,
+				message_type: 'Reset password',
+				public_api: false,
+			});
+			if (error instanceof Error) {
+				throw new ResponseHelper.ResponseError(
+					`Please contact your administrator: ${error.message}`,
+					undefined,
+					500,
+				);
+			}
+		}
+
+		Logger.info('Sent password reset email successfully', { userId: user.id, email });
+		void InternalHooksManager.getInstance().onUserTransactionalEmail({
+			user_id: id,
+			message_type: 'Reset password',
+			public_api: false,
+		});
+
+		void InternalHooksManager.getInstance().onUserPasswordResetRequestClick({
+			user_id: id,
+		});
+	}
+
+	/**
+	 * Verify password reset token and user ID.
+	 *
+	 * Authless endpoint.
+	 */
+	@Get('/resolve-password-token')
```

Comment:
yeah. I don't know what to do about these. I could just stick with the naming format enforced by our linting rules, but those seem disabled in so many places that I'm not sure if that's the naming convention we want to follow.
Has there been any dicussions on making casing consistent, and removing all the `eslint-disable @typescript-eslint/naming-convention` ?

Reply:
For context, our ESLint rules came in after most of the existing code had been written, so most preexisting code was exempted and needs to be adjusted, while later additions should (ideally) not be disabling them.

That said, the lint rules are always up for discussion - we can even set special rules for decorators using the `custom` option for `naming-convention`, so it's up to us to decide how we want this. Personally I think `@GET` is the clearest and easiest to remember.

Reply:
That makes sense. I'd went with PascalCase here because I noticed that was the naming convention on some other projects, like NestJS and Angular. 
So, how do we get a consensus on this from a larger team? and do we want to then wait for that, or should we merge this as it is for now, and open a PR to make all naming consistent across at least this package.

Reply:
Agree, simpler to tackle it separately. Feedback on linting historically has been rather low, so I think the easiest would be propose what you think is best and ask for views with a deadline.